internal/internal_workflow.go

```diff
@@ -513,6 +513,7 @@ func (d *syncWorkflowDefinition) Execute(env WorkflowEnvironment, header *common
 			// we are yielding.
 			state := getState(d.rootCtx)
 			state.yield("yield before executing to setup state")
+			state.unblocked()
```

Comment:
@cretz any backwards compatibility concerns about this unblock? Based on my understand of the scheduler this unblock call should always have been here, but it didn't matter till the other changes in this PR.

Reply:
I admit not understanding much of the behavior here. But I would think since this all happens before user code, there should be no affect right? However, the note in the `yield` makes me think that there are other coroutines setting up state that come back and unblock this one. I think to feel confident here, we're going to have to find everything this is yielding to.

Reply:
> However, the note in the yield makes me think that there are other coroutines setting up state that come back and unblock this on

as far as I am aware this is not true, yields don't need to be "unblocked". My concern is this `yield` without a block was putting the scheduler in a bad state on the first WFT that was never exposed before. The question I am struggling to answer is if this bad state actually effected any user code.

Reply:
> yield before executing to setup state

^ That's my bigger concern. What's it yielding to? And can what it's yielding to start its own coroutines that need to be added to the list while this one remains blocked? In other words, why does this yield at all?

Reply:
What's it yielding to is not a concern because I am not removing the yield if anything does. I don't actually think it does yield to anything. Maybe a workflow interceptor could spawn a workflow goroutine? I am more concerned with the coroutines state on the first WFT.

Reply:
> What's it yielding to is not a concern because I am not removing the yield if anything does.

If it yields to something that itself starts a coroutine, does marking this as unblocked mean this may run before that other coroutine where it didn't before? Now that I think about it, if it's not waiting to be unblocked I guess it doesn't matter. If there are no concerns that whatever it is yielding to will now have its order affected by this call, I'm good w/ this.

Reply:
Yeah after talking it over I think it is safe